src/BlockTypes/ProductQuery.php

```diff
@@ -115,8 +115,9 @@ public function update_query( $pre_render, $parsed_block ) {
 	public function update_rest_query( $args, $request ) {
 		$on_sale_query = $request->get_param( '__woocommerceOnSale' ) !== 'true' ? array() : $this->get_on_sale_products_query();
 		$orderby_query = $this->get_custom_orderby_query( $request->get_param( 'orderby' ) );
+		$tax_query     = $this->get_product_attributes_query( $request->get_param( '__woocommerceAttributes' ) );
 
-		return array_merge( $args, $on_sale_query, $orderby_query );
+		return array_merge( $args, $on_sale_query, $orderby_query, $tax_query );
```

Comment:
Reviewing this make me realize that I can do the same for #7682 as long as we only have on `meta_query` ( for `get_product_attributes_query()` ) and one `tax_query` (for `get_product_attributes_query()`). And we should keep this simple implementation until we can not use `array_merge` anymore, in the other worlds, until we need to merge more than one `meta_query`/`tax_query`. What do you think?

Reply:
Totally agree: the simpler, the better.